lib/reek/smells/smell_detector.rb

```diff
@@ -15,6 +15,7 @@ module Smells
     # :reek:TooManyMethods: { max_methods: 19 }
     # :reek:TooManyInstanceVariables: { max_instance_variables: 5 }
     class SmellDetector
+      private_attr_reader :config
```

Comment:
Let's put calls to class macro's always at the top. I find it highly confusing that they are sometimes scattered across the file.